FreeRTOSv202112.00/FreeRTOS/Demo/ARM7_LPC2129_Keil_RVDS/Task2 Files/source/task2.c

```diff
@@ -0,0 +1,94 @@
+#include "task2.h"
+#include "FreeRTOS.h"
+#include "semphr.h"
+#include "serial.h"
+#include "LED.h"
+
+
+SemaphoreHandle_t Semaphore_Handler = NULL;
+
+TaskHandle_t Task_1_TaskHandle = NULL;					// Task 1 Task Handler
+TaskHandle_t Task_2_TaskHandle = NULL;					// Task 2 Task Handler 
+
+void vTask_1(void* pvParameters)
+{
+	static unsigned char semaphore_state = pdFALSE ;
+	static unsigned char string_counter = 0;
+	static unsigned int heavy_load_counter = 0;
+	for(;;)
+	{
+		semaphore_state = xSemaphoreTake( Semaphore_Handler, ( TickType_t ) 0 );
+		if(semaphore_state == pdTRUE)
+		{
+			led_on(2);
+			for(string_counter = 0 ; string_counter < 10 ; string_counter++)
+			{
+				for (heavy_load_counter = 0; heavy_load_counter < 100000; heavy_load_counter++ )
```

Comment:
you shouldn't  make heavy load in this task

Reply:
I solved it